src/report.rs

```diff
@@ -17,8 +17,15 @@ pub trait Reporter<P: Package, VS: VersionSet> {
     type Output;
```

Comment:
I don't think this trait pulls its weight, I think users would be just as well served by freestanding functions. Do you want to do that in this PR or leave it for follow-up?

Reply:
I agree this trait is a little weird unless users are expected to be passing around generic reporters. It makes a bit more sense now that there's another method, but I agree it seems sensible to remove. I'd prefer to track this in a new issue.